js/main.js

```diff
@@ -0,0 +1,134 @@
+// dom queries
+const $searchBar = document.querySelector('#query');
+const $apiEndpoint = 'https://openlibrary.org/search.json?q=';
+const limit = '&limit=10&offset=0';
+const $icon = document.querySelector('.icon');
+// const $fwrdArrow = document.querySelector('.forward-arrow');
+const $backArrow = document.querySelector('.back-arrow');
+const $arrowContainer = document.querySelector('.arrow-container');
+const $entriesTable = document.querySelector('.entries-table');
+// const $entryView = document.querySelector('[data-view=entries]');
+// const $formView = document.querySelector('[data-view=entry-form]');
+// const $form = document.querySelector('#form');
+const searchUrl = 'https://covers.openlibrary.org/b/id/';
+const urlSuffix = '.jpg';
+
+// array to store search results
+const currentResults = [];
+
+// function handles search query from open library API
+function getBookData(event) {
+  // prevents default form reset behavior
+  event.preventDefault();
+  // gets string from search bar, splits it, adds '+' between each character and concatenates w/ api endpoint
+  let searchTerms = $searchBar.value;
+  const splitTerms = searchTerms.split(' ');
+  searchTerms = splitTerms.join('+');
+  const requestUrl = $apiEndpoint + searchTerms + limit;
+  // creates new xhr object
+  const xhr = new XMLHttpRequest();
+  // sets request method and URL
+  xhr.open('GET', requestUrl);
+  // gets body of HTTP response once it has been converted from JSON to JS objects
+  xhr.responseType = 'json';
+  // executes function when response is eventually loaded
+  xhr.addEventListener('load', function () {
+    // handles API response
+    const response = xhr.response;
+    // access and use key-value pairs
+    const results = response.docs;
+    //  removes existing entries from the table before next search
+    const existingEntries = document.querySelectorAll('.results-row');
+
+    // initiates loop that iterates over each book object in the results array obtainedfrom the API response
+    existingEntries.forEach(function (entry) {
+      entry.remove();
+    });
+    // assigns api search results to variables for later use
+    results.forEach(function (book) {
+      const title = book.title;
+      const author = book.author_name;
+      const cover = book.cover_i;
+      const hersheyUrl = searchUrl + cover + urlSuffix;
+      // conditional statement filters out incomplete search results and behaves normally for results with all info present
+      if (cover === undefined || title === undefined || author === undefined) {
+        // eslint-disable-next-line no-console
+        console.log('hello world');
```

Comment:
Remove all `console.log`, they are only for testing and debugging. Also don't add `eslint` rules without asking me first